### PR TITLE
vNext controller visualization profile visualization manager

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# Overview
Updated implementation, moving the Visualization system to it's own manager.

It's a lot cleaner imho, removes the dependency on the Input System. All tested and working fine.
Also moved the Visualization parts to their own feature folders, so it's clear it is separate.

# Changes
- Adds - MixedRealityVisualizerManager (and relevant interfaces)
- Device layer simply requests for a Visualizer on start (of configured)
- Visualizer handles it's destruction properly and removes itself from the detected visualizers in the manager
- Retained Controller reference in the visualizer so it knows what it's attached to.

# Debate
I changed all references for "ControllerVisualizer" to just "Visualizer" as almost any input (or thing) could be Visualised.  Do we agree with that, or should we revert that part?

### CODE REVIEW

File: Assets/MixedRealityToolkit-SDK/Inspectors/Input/Handlers/MixedRealityControllerVisualizerInspector.cs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-using Microsoft.MixedReality.Toolkit.SDK.UX.Controllers;
+using Microsoft.MixedReality.Toolkit.SDK.VisualizationSystem;
 using UnityEditor;
 
 namespace Microsoft.MixedReality.Toolkit.SDK.Inspectors.Input.Handlers
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Inspectors/Input/Handlers/MixedRealityControllerVisualizerInspector.cs]: Why is this still in the input namespace if you made it, it's own manager?

File: Assets/MixedRealityToolkit/_Core/Definitions/Visualization/MixedRealityVisualizationProfile.cs [hunk 2]
```diff
@@ -26,17 +27,31 @@ public bool RenderMotionControllers
...
-            get { return controllerVisualizationType; }
-            private set { controllerVisualizationType = value; }
+            get { return visualizationManager; }
+            private set { visualizationManager = value; }
+        }
+
+        [SerializeField]
+        [Implements(typeof(IMixedRealityVisualizer), TypeGrouping.ByNamespaceFlat)]
+        [Tooltip("The concrete Visualizer component to use on the rendered controller model.")]
+        private SystemType visualizerType;
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Definitions/Visualization/MixedRealityVisualizationProfile.cs:46]: Why did you remove this from the profile, where it belongs?

File: Assets/MixedRealityToolkit-SDK/Features/Visualization/MixedRealityVisualizer.cs [hunk 3]
```diff
@@ -3,21 +3,43 @@
...
+
         /// <inheritdoc />
-        public GameObject ControllerGameObjectReference => gameObject;
+        public GameObject VisualizerGameObjectReference => gameObject;
+
+        public override void OnSourceLost(SourceStateEventData eventData)
+        {
+            if (eventData.Controller?.ControllerHandedness == Handedness)
+            {
+                visualizationManager?.DetectedVisualizers.Remove(this);
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Features/Visualization/MixedRealityVisualizer.cs]: Do we also want to destroy the game obhect and clean up the scene?

File: Assets/MixedRealityToolkit-SDK/Features/Visualization/MixedRealityVisualizer.cs [hunk 4]
```diff
@@ -3,21 +3,43 @@
...
+            set { visualizationManager = value; }
+        }
+
         /// <inheritdoc />
-        public GameObject ControllerGameObjectReference => gameObject;
+        public GameObject VisualizerGameObjectReference => gameObject;
+
+        public override void OnSourceLost(SourceStateEventData eventData)
+        {
+            if (eventData.Controller?.ControllerHandedness == Handedness)
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Features/Visualization/MixedRealityVisualizer.cs]: Shouldn't we also check if it's the same controller? Not just handedness?

File: Assets/MixedRealityToolkit-SDK/Profiles/DefaultMixedRealityRegisteredComponentsProfile.asset [hunk 5]
```diff
@@ -54,3 +54,9 @@ MonoBehaviour:
     priority: 10
     runtimePlatform: -1
     configurationProfile: {fileID: 0}
+  - componentType:
+      reference: 
+    componentName: New Configuration 6
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Profiles/DefaultMixedRealityRegisteredComponentsProfile.asset]: What's this?

File: Assets/MixedRealityToolkit-SDK/Features/Visualization/MixedRealityVisualizationManager.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+            base.Initialize();
+            InitializeInternal();
+        }
+
+        /// <summary>
+        /// Performs initialization tasks for the BoundaryManager.
+        /// </summary>
+        private void InitializeInternal()
+        {
+
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Features/Visualization/MixedRealityVisualizationManager.cs]: We need to remove all the boundary language and any unused methods. A good cleanup here would be nice

File: Assets/MixedRealityToolkit/_Core/Definitions/Visualization/MixedRealityVisualizationProfile.cs (hunk 2 above)
Comment (SimonDarksideJ) [reply Assets/MixedRealityToolkit/_Core/Definitions/Visualization/MixedRealityVisualizationProfile.cs:46]: It was renamed, not removed.

Instead of ControllerVisualizationType ( The scene visualiser to use) , you have now:
* VisualizationManager - The manager to use
* VisualizerType - The scene visualiser to use
